plugins/solaris/io_disk

```diff
@@ -0,0 +1,301 @@
+#!/bin/bash
+
+: << =cut
+
+=head1 NAME
+
+  io_disk - Multigraph plugin to monitor disks for Solaris.
+
+  These functions are implemented:
+     ops     : similar to iostat r/s, w/s
+     bytes   : similar to iostat kr/s, kw/s 
+     busy    : similar to iostat %b, %w
+     queue   : similar to iostat actv, wait
+     latency : similar to iostat asvc_t, wsvc_t
+
+  Any device found in /usr/bin/kstat can be monitored.
+
+  Tested with Solaris 10 and 11, and it should work with Solaris 8, 9.
+
+=head1 CONFIGURATION
+
+  Make symlink:
+    cd /path/to/munin/etc/plugins
+    ln -s /path/to/munin/lib/plugins/io_disk .
+
+  The RRD files generated by io_busy_, io_ops_, io_bytes_ can be taken over 
+  by this plugin.
+  Thus, please remove symlinks of those plugins before using this plugin.
+
+  By default, this plugin monitors disk devices. And also it can monitor 
+  NFS and Tape devices as much as io_* plugins with setting environments.
+
+=head1 ENVIRONMENT VARIABLES
+
+  env.class - kstat class. See man kstat -c option.
+    example:  env.class  /disk|nfs|tape/
+    default:  disk
+
+  env.module - Module name. Only used in internal graph name.  If you would not
+    to like to take over RRDs generated by io_*_sd plugins, please change it 
+    to another name.
+
+    example:  env.module  something
+    default:  sd
+
+  env.module_regex - kstat module. See man kstat -m option.
+    example:  env.module_regex  /^(s?sd|vdc|zvblk|dad|md|nfs|tape)$/
+    default:  /^(s?sd|vdc|zvblk|dad|md)$/
+
+  env.title_type - Device type shown in graph title.
+    example:  env.title_type  Disk Device, NFS, Tape
+    default:  Disk Device
+
+  env.exclude - Device instance name(s) to exclude seperated by white-space.
+    example:  env.exclude  sd0 ssd1
+    default:  none
+
+  env.graph_width - Graph width in pixel
+    example:  env.graph_width  450
+    default:  none (usually 400)
+
+=head1 AUTHOR
+
+  Unknown Author
+  Improved by K.Cima https://github.com/shakemid
+
+=head1 LICENSE
+
+  GPLv2
+
+=head1 MAGIC MARKERS
+
+  #%# family=auto
+  #%# capabilities=autoconf
+
+=cut
+
+# Include plugin.sh
+. "${MUNIN_LIBDIR:-}/plugins/plugin.sh"
+is_multigraph "$@"
+
+# Shell options
+set -o nounset
+set -o pipefail
```

Comment:
It enables to get return code ($?) when command fails in front of pipe. (e.g bad_command | good_command; echo $?)
Without -o pipefail, shell returns status of good_command (0). 
But, I didn't use it in this plugin. Is it better to remove it?